### ISSUE TITLE
[BUG] eeprom not shoing all settings marlin 2.0

### ISSUE THREAD

**Issue by jon012198** [MarlinFirmware/Marlin#13333]
it is only showing info steps filament advanced 
here is my configs
[Marlin.zip](https://github.com/MarlinFirmware/Marlin/files/2946418/Marlin.zip)
i am running marlin 2.0
![firefox_2019-03-08_10-37-26](https://user-images.githubusercontent.com/7279835/54038332-47f65c00-418e-11e9-8eaf-625bff579173.png)
![firefox_2019-03-08_10-37-36](https://user-images.githubusercontent.com/7279835/54038333-47f65c00-418e-11e9-84df-40b13e977751.png)
![firefox_2019-03-08_10-37-51](https://user-images.githubusercontent.com/7279835/54038335-47f65c00-418e-11e9-9fb2-4aa99bdc151b.png)
![firefox_2019-03-08_10-37-08](https://user-images.githubusercontent.com/7279835/54038337-47f65c00-418e-11e9-839b-1e3836f83f69.png)

**Comment by RudyBenner** [comment]
I am also having problems with EEPROM not displaying all the settings.
When I connect with a terminal program (Putty) and do M501
the output looks like it may have extra line feed characters.
I have reverted to several older versions with the same result.
Yet, I have no problem with the new SBASE board.
I will attach some files.
Pronterface also shows the same botched up formatting.

I use Repetier Host

What is the correct EOL sequence? My VSCode shows LF, I have not changed anything here.

[BUG.zip](https://github.com/MarlinFirmware/Marlin/files/2947640/BUG.zip)

**Comment by kAdonis** [comment]
Yes, I see this behavior

> like it may have extra line feed characters

in commit from 9. March
M501 is showing every parameter, but is difficult to read
I flashed marlin from 3. March and M501 is working as expected
(Ramps/Re-Arm)

**Comment by jon012198** [comment]
Ok I am glad I am not the only one

**Comment by RudyBenner** [comment]
Yesterday it was fine.

**Comment by kAdonis** [comment]
@thinkyhead The cause for this issue is in #13311 in serial.h line 111
The new SERIAL_ECHOLNPAIR(...) sends an EOL after **every** pair
That is probably not as it should be, but I dont find an elegant way to fix this

**Comment by p3p** [comment]
I can barely parse thinkyheads magic macros but I think moving the EOL to the top level macro should fix it, 

`serial.h line 135`
```c++
#define SERIAL_ECHOLNPAIR(...) do{ _SELP_N(NUM_ARGS(__VA_ARGS__),__VA_ARGS__); SERIAL_EOL(); }while(0)
```

**Comment by RudyBenner** [comment]
Thank you sir, we seem to be working again. 
rudy

**Comment by fape** [comment]
I got compile errors:
Latest marlin2.0, latest platformio. Only changes motherboard to BOARD_BIGTREE_SKR_V1_3 and serial1 to -1.
Edit: same with default config.

```
In file included from Marlin\src\inc/MarlinConfig.h:49,
from Marlin\src\Marlin.h:24,
from Marlin\src\Marlin.cpp:31:
Marlin\src\Marlin.cpp: In function 'void setup()':
Marlin\src\inc/../core/serial.h:115:64: error: expected 'while' before ';' token
#define _SELP_4(a,b,...)  do{ _SEP_2(a,b); _SELP_2(__VA_ARGS__); }while(0)
^

Marlin\src\inc/../core/serial.h:110:27: note: in expansion of macro '_SELP_4'
#define __SELP_N(N,...)   _SELP_##N(__VA_ARGS__)
^~~~~~
Marlin\src\inc/../core/serial.h:111:27: note: in expansion of macro '__SELP_N'
#define _SELP_N(N,...)    __SELP_N(N,__VA_ARGS__)
^~~~~~~~
Marlin\src\inc/../core/serial.h:137:32: note: in expansion of macro '_SELP_N'
#define SERIAL_ECHOLNPAIR(...) _SELP_N(NUM_ARGS(__VA_ARGS__),__VA_ARGS__)
^~~~~~~
Marlin\src\Marlin.cpp:913:3: note: in expansion of macro 'SERIAL_ECHOLNPAIR'
SERIAL_ECHOLNPAIR(MSG_FREE_MEMORY, freeMemory(), MSG_PLANNER_BUFFER_BYTES, (int)sizeof(block_t) * (BLOCK_BUFFER_SIZE));
```

**Comment by RudyBenner** [comment]
This error was corrected, unfortunately, he made some more changes and broke it.
Patience.

**Comment by fape** [comment]
I know. I fixed it in #13343

**Comment by RudyBenner** [comment]
I reverted that file. Now it compiles.
